storage/integration_test.go

```diff
@@ -2250,6 +2250,75 @@ func TestIntegration_WriterContentType(t *testing.T) {
 	})
 }
 
+func TestIntegration_WriterChunksize(t *testing.T) {
+	ctx := skipJSONReads(skipGRPC("https://github.com/googleapis/google-cloud-go/issues/7798"), "no reads in test")
+	multiTransportTest(ctx, t, func(t *testing.T, ctx context.Context, bucket, _ string, client *Client) {
+		obj := client.Bucket(bucket).Object("writer-chunksize-test" + uidSpaceObjects.New())
+		objSize := 17 << 10 << 10 // 17 Mib
+		contents := bytes.Repeat([]byte("a"), objSize)
+
+		for _, test := range []struct {
+			desc             string
+			chunksize        int
+			wantBytesPerCall int64
+		}{
+			{
```

Comment:
Let's also check that progressfunc is called the expected number of times for each case.

Reply:
Good call, adding that uncovered a discrepancy we have between HTTP and GRPC - the progress function is only called in HTTP if the upload is resumable (so it is not called at all for a one-shot), in GRPC the function is called at least once even if the upload is not resumable. 

The docs for the progress func do specify it gets called only if the upload requires more than one call, so I assume we want to modify gRPC to comply with this?

Reply:
Agreed, that sounds right.

Reply:
Opened https://github.com/googleapis/google-cloud-go/issues/7839 to keep track